scripts/curl_install_pypi/install.py

```diff
@@ -77,69 +118,216 @@ def create_virtualenv(tmp_dir, version, install_dir):
     exec_command(cmd, cwd=working_dir)
 
 def install_cli(install_dir, tmp_dir):
-    path_to_pip = os.path.join(install_dir, BIN_DIR_NAME, 'pip')
+    path_to_pip = os.path.join(install_dir, 'bin', 'pip')
     cmd = [path_to_pip, 'install', '--cache-dir', tmp_dir, 'azure-cli', '--upgrade']
     exec_command(cmd)
 
 def create_executable(exec_dir, install_dir):
     create_dir(exec_dir)
-    exec_filename = os.path.join(exec_dir, EXECUTABLE_NAME)
-    with open(exec_filename, 'w') as exec_file:
-        exec_file.write(AZ_DISPATCH_TEMPLATE.format(
-            install_dir=install_dir,
-            bin_dir_name=BIN_DIR_NAME))
-    cur_stat = os.stat(exec_filename)
-    os.chmod(exec_filename, cur_stat.st_mode | stat.S_IXUSR | stat.S_IXGRP | stat.S_IXOTH)
-    return exec_filename
-
-def prompt_input(message):
-    return None if DISABLE_PROMPTS else input(message)
+    exec_filepath = os.path.join(exec_dir, EXECUTABLE_NAME)
+    with open(exec_filepath, 'w') as exec_file:
+        exec_file.write(AZ_DISPATCH_TEMPLATE.format(install_dir=install_dir))
+    cur_stat = os.stat(exec_filepath)
+    os.chmod(exec_filepath, cur_stat.st_mode | stat.S_IXUSR | stat.S_IXGRP | stat.S_IXOTH)
+    print_status("The executable is available at '{}'.".format(exec_filepath))
+    return exec_filepath
 
 def get_install_dir():
-    prompt_message = 'In what directory would you like to place the install? (leave blank to use {}): '.format(DEFAULT_INSTALL_DIR)
-    install_dir = prompt_input(prompt_message) or DEFAULT_INSTALL_DIR
-    install_dir = os.path.realpath(os.path.expanduser(install_dir))
-    if not os.path.isdir(install_dir):
-        print("Directory '{}' does not exist. Creating directory...".format(install_dir))
-        create_dir(install_dir)
-    print("We will install at '{}'.".format(install_dir))
+    install_dir = None
+    while not install_dir:
+        prompt_message = 'In what directory would you like to place the install?'
+        install_dir = prompt_input_with_default(prompt_message, DEFAULT_INSTALL_DIR)
+        install_dir = os.path.realpath(os.path.expanduser(install_dir))
+        if ' ' in install_dir:
+            print_status("The install directory '{}' cannot contain spaces.".format(install_dir))
+            install_dir = None
+        else:
+            create_dir(install_dir)
+            if os.listdir(install_dir):
+                print_status("'{}' is not empty and may contain a previous installation.".format(install_dir))
+                ans_yes = prompt_y_n('Remove this directory?', 'n')
```

Comment:
Is just removing the directory sufficient? I know it is for a virtual environment install.